homeassistant/components/iaqualink/__init__.py

```diff
@@ -151,15 +169,16 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     async def _async_systems_update(now):
         """Refresh internal state for all systems."""
-        prev = systems[0].last_run_success
-
-        await systems[0].update()
-        success = systems[0].last_run_success
+        prev = systems[0].online
 
-        if not success and prev:
-            _LOGGER.warning("Failed to refresh iAqualink state")
-        elif success and not prev:
-            _LOGGER.warning("Reconnected to iAqualink")
+        try:
+            await systems[0].update()
+        except AqualinkServiceException as svc_exception:
+            _LOGGER.warning("Failed to refresh iAqualink state: %s", svc_exception)
+        else:
+            cur = systems[0].online
+            if cur is True and prev is not True:
+                _LOGGER.warning("Reconnected to iAqualink")
 
         async_dispatcher_send(hass, DOMAIN)
```

Comment:
If they have multiple systems this dispatcher will trigger all of them.  It should include something unique like the config entry id

Reply:
Nevermind, you are limited to only one instance with `single_instance_allowed`

Reply:
Right now the integration only supports the first system instance returned by the library so this is a non-issue.